petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx

```diff
@@ -0,0 +1,191 @@
+    if (!product.productName) {
+      newError.productName = 'Product name is required';
+    }
+    if (!product.productDescription) {
+      newError.productDescription = 'Product description is required';
+    }
+    if (!product.productPrice) {
+      newError.productPrice = 'Product price is required';
+    }
+    if (!product.productQuantity) {
+      newError.productQuantity = 'Product quantity is required';
+    }
+    if (!product.productSalePrice) {
+      newError.productSalePrice = 'Product sale price is required';
+    }
+    setError(newError);
+    return Object.keys(newError).length === 0;
+  };
+
+  const handleSubmit = async (
+    event: FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    event.preventDefault();
+    if (!validate()) return;
+
+    setLoading(true);
+    setErrorMessage('');
+    setSuccessMessage('');
+    setShowNotification(false);
+
+    try {
+      if (inventoryId && productId) {
+        await updateProductInInventory(inventoryId, productId, product);
+        setSuccessMessage('Product updated successfully');
+        setShowNotification(true);
+        setTimeout(() => {
+          navigate(`/inventory/${inventoryId}/products`);
+        }, 2000);
+      }
+    } catch (error) {
+      const apiError = error as ApiError;
+      setErrorMessage(`Error updating product: ${apiError.message}`);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Handle input changes
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setProduct({
+      ...product,
+      [e.target.name]:
+        e.target.type === 'number'
+          ? parseFloat(e.target.value)
+          : e.target.value,
+    });
+  };
+
+  return (
+    <div>
+      <h1>Edit Product</h1>
+      {error && <p style={{ color: 'red' }}>{error.message}</p>}
+
+      <form onSubmit={handleSubmit}>
+        <div>
+          <label>Product Name</label>
+          <input
+            type="text"
+            name="productName"
+            value={product.productName}
+            onChange={handleChange}
+            required
+          />
+        </div>
+
+        <div>
+          <label>Product Description</label>
+          <input
+            type="text"
+            name="productDescription"
+            value={product.productDescription}
+            onChange={handleChange}
+            required
+          />
+        </div>
+
+        <div>
+          <label>Product Price</label>
+          <input
+            type="number"
+            name="productPrice"
+            value={product.productPrice}
+            onChange={handleChange}
+            required
+          />
+        </div>
+
+        <div>
+          <label>Product Quantity</label>
+          <input
+            type="number"
+            name="productQuantity"
+            value={product.productQuantity}
+            onChange={handleChange}
+            required
+          />
+        </div>
+
+        <div>
```

Comment:
Great but I tkink we don;t use the sale price anymore maybe change this later

Reply:
Ok noted thanks